hypothesis-python/docs/details.rst

```diff
@@ -543,3 +543,65 @@ changes between Python 3.5.0 and 3.6.1, including at minor versions.  These
 are all supported on a best-effort basis, but you may encounter problems with
 an old version of the module.  Please report them to us, and consider
 updating to a newer version of Python as a workaround.
+
+
+.. _our-type-hints:
+
+------------------------------
+Type Annotations in Hypothesis
+------------------------------
+
+If you install Hypothesis and use :pypi:`mypy` 0.590+, or another
+:PEP:`561`-compatible tool, the type checker should automatically pick
+up our type hints.
+
+.. note::
+    Hypothesis' type hints may make breaking changes between minor releases.
```

Comment:
Should the paragraphs below be part of this note? They qualify it, so I think it makes sense to have them logically grouped.

It also might be nice to make clear that we do plan to stabilise it eventually (but not before the main typing module becomes non-provisional)